### DISCUSSION TITLE
PitchShift Effect

### DISCUSSION THREAD

**Discussion by weow147** [pschatzmann/arduino-audio-tools#1248] [Q&A] [answered]
Does [PitchShift ](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_pitch_shift.html) use RingBuffer by default or do i need to specify it somehow?

On a related note, i'm getting a warning with shift values less than 1.0 with i2s buffer settings:
DEFAULT_BUFFER_SIZE = 256
I2S_BUFFER_SIZE = 256

`PitchShift.h : 372 - phase allign failed: maybe the buffer is too small`

My application is real-time DSP.  any larger than buffer 256 and the delay is unacceptable.

The [wiki ](https://github.com/pschatzmann/arduino-audio-tools/wiki/Pitch-Shifting) indicates that the ringbuffer is free of flaws?

**Comment by pschatzmann** [comment, marked as answer]
If you use the PitchShift effect class, you can see in the class documentation that it is using the VariableSpeedRingBuffer.
In the constructor you can indicate the ring buffer size!

**Reply by weow147** [reply]
Perfect, thanks!
